wdl-engine/src/value.rs

```diff
@@ -2052,6 +2590,339 @@ impl From<Struct> for CompoundValue {
     }
 }
 
+/// Represents a value for `task` variables in WDL 1.2.
+///
+/// Task values are cheap to clone.
+#[derive(Debug, Clone)]
+pub struct TaskValue {
+    /// The name of the task.
+    name: Arc<String>,
+    /// The id of the task.
+    id: Arc<String>,
+    /// The container of the task.
+    container: Option<Arc<String>>,
+    /// The allocated number of cpus for the task.
+    cpu: f64,
+    /// The allocated memory (in bytes) for the task.
+    memory: i64,
+    /// The GPU allocations for the task.
+    ///
+    /// An array with one specification per allocated GPU; the specification is
+    /// execution engine-specific.
+    gpu: Array,
+    /// The FPGA allocations for the task.
+    ///
+    /// An array with one specification per allocated FPGA; the specification is
+    /// execution engine-specific.
+    fpga: Array,
+    /// The disk allocations for the task.
+    ///
+    /// A map with one entry for each disk mount point.
+    ///
+    /// The key is the mount point and the value is the initial amount of disk
+    /// space allocated, in bytes.
+    disks: Map,
+    /// The current task attempt count.
+    ///
+    /// The value must be 0 the first time the task is executed and incremented
+    /// by 1 each time the task is retried (if any).
+    attempt: i64,
+    /// The time by which the task must be completed, as a Unix time stamp.
+    ///
+    /// A value of `None` indicates there is no deadline.
+    end_time: Option<i64>,
+    /// The task's return code.
+    ///
+    /// Initially set to `None`, but set after task execution completes.
+    return_code: Option<i64>,
+    /// The task's `meta` section as an object.
+    meta: Object,
+    /// The tasks's `parameter_meta` section as an object.
+    parameter_meta: Object,
+    /// The task's extension metadata.
+    ext: Object,
```

Comment:
What's this? Something added in 1.2 that I'm not familiar with?

Reply:
It's from 1.2's section on [runtime access to task information](https://github.com/openwdl/wdl/blob/wdl-1.2/SPEC.md#runtime-access-to-requirements-hints-and-metadata).